tools/pubsys/src/repo.rs

```diff
@@ -0,0 +1,592 @@
+
+    let key_source: Box<dyn KeySource> = match signing_key_config {
+        SigningKeyConfig::file { path } => Box::new(LocalKeySource { path: path.clone() }),
+        SigningKeyConfig::ssm { parameter } => Box::new(SsmKeySource {
+            profile: None,
+            parameter_name: parameter.clone(),
+            key_id: None,
+        }),
+    };
+
+    let signed_repo = editor.sign(&[key_source]).context(error::RepoSign)?;
+
+    // Write repo   =^..^=   =^..^=   =^..^=   =^..^=
+
+    let metadata_out_dir = repo_args.outdir.join("metadata");
+    let targets_out_dir = repo_args.outdir.join("targets");
+    fs::create_dir_all(&metadata_out_dir).context(error::CreateDir {
+        path: &metadata_out_dir,
+    })?;
+    fs::create_dir_all(&targets_out_dir).context(error::CreateDir {
+        path: &targets_out_dir,
+    })?;
+
+    info!("Writing repo to: {}", repo_args.outdir.display());
+    signed_repo
+        .write(&metadata_out_dir)
+        .context(error::RepoWrite {
+            path: &repo_args.outdir,
+        })?;
+
+    // Copy manifest with proper name instead of tempfile name
+    debug!("Copying manifest.json into {}", targets_out_dir.display());
+    signed_repo
+        .copy_target(&manifest_path, &targets_out_dir, Some("manifest.json"))
+        .context(error::CopyTarget {
+            target: &manifest_path,
+            path: &targets_out_dir,
+        })?;
+
+    // Copy / link any other user requested targets
+    for copy_target in copy_targets {
+        debug!(
+            "Copying target '{}' into {}",
+            copy_target.display(),
+            targets_out_dir.display()
+        );
+        signed_repo
+            .copy_target(copy_target, &targets_out_dir, None)
+            .context(error::CopyTarget {
+                target: copy_target,
+                path: &targets_out_dir,
+            })?;
+    }
+    for link_target in link_targets {
+        debug!(
+            "Linking target '{}' into {}",
+            link_target.display(),
+            targets_out_dir.display()
+        );
+        signed_repo
+            .link_target(link_target, &targets_out_dir, None)
+            .context(error::LinkTarget {
+                target: link_target,
+                path: &targets_out_dir,
+            })?;
+    }
+
+    Ok(())
+}
+
+mod error {
+    use snafu::Snafu;
+    use std::io;
+    use std::path::PathBuf;
+    use url::Url;
+
+    #[derive(Debug, Snafu)]
+    #[snafu(visibility = "pub(super)")]
+    pub(crate) enum Error {
+        #[snafu(display("Failed to add new update to manifest: {}", source))]
+        AddUpdate {
+            source: update_metadata::error::Error,
+        },
+
+        #[snafu(display("Failed to add new target '{}' to repo: {}", path.display(), source))]
+        AddTarget {
+            path: PathBuf,
+            source: tough::error::Error,
+        },
+
+        #[snafu(display("Failed to build Target from path '{}': {}", path.display(), source))]
```

Comment:
nit:
```suggestion
        #[snafu(display("Failed to build target from path '{}': {}", path.display(), source))]
```

Reply:
The "T" was there to represent it's building a Target object from that path, but I agree with what I think you're getting at, which is that a user shouldn't have to understand that, or even question why it's capitalized.  I'm going to use "target metadata" as a clearer version.